Content.Server/Chemistry/EntitySystems/SolutionPurgeSystem.cs

```diff
@@ -0,0 +1,60 @@
+using Content.Server.Chemistry.Components;
+using Content.Server.Chemistry.Components.SolutionManager;
+using Content.Shared.FixedPoint;
+using Robust.Shared.Timing;
+
+namespace Content.Server.Chemistry.EntitySystems;
+
+public sealed class SolutionPurgeSystem : EntitySystem
+{
+    [Dependency] private readonly SolutionContainerSystem _solutionContainer = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<SolutionPurgeComponent, EntityUnpausedEvent>(OnUnpaused);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var query = EntityQueryEnumerator<SolutionPurgeComponent, SolutionContainerManagerComponent>();
+        while (query.MoveNext(out var uid, out var purge, out var manager))
+        {
+            if (_timing.CurTime < purge.NextPurgeTime)
+                continue;
+
+            // timer ignores if it's empty, it's just a fixed cycle
+            purge.NextPurgeTime = _timing.CurTime + purge.Duration;
+            if (_solutionContainer.TryGetSolution(uid, purge.Solution, out var solution, manager))
+            {
+                // purge.Quantity is the *total* quantity of reagent we want to remove.
+                // We need to calculate how much *per reagent*, excluding reagents that
+                // must be reserved.
+                int reagentsToRemove = solution.Contents.Count;
+                foreach (var reagentId in purge.Preserve)
+                    if (solution.ContainsReagent(reagentId))
+                        reagentsToRemove--;
+
+                if (reagentsToRemove != 0)
+                {
+                    FixedPoint2 quantityPerReagent = purge.Quantity / reagentsToRemove;
+                    foreach (var reagent in solution.Contents.ToArray())
+                    {
+                        if (purge.Preserve.Contains(reagent.ReagentId))
+                            continue;
+                        _solutionContainer.TryRemoveReagent(uid, solution, reagent.ReagentId, quantityPerReagent);
+                    }
+                }
```

Comment:
You should probably just update SplitSolutionWithout to take params string[] instead.

Reply:
I took a stab at a solution[^1] in a2bffd7. I'm not 100% convinced it's the best approach for this particular codebase, so feel free to have a gander.

[^1]: Pun absolutely intended.